### ISSUE TITLE
[dv] Enable using different OTP image 

### ISSUE THREAD

**Issue by sha-ron** [lowRISC/opentitan#9229]
Currently the OTP image that is used for the chip level tests is one that matches the open source and doesn’t fit the foundry OTP. Need an option to use a different image for the foundry OTP.
CC: @tjaychen @ankamate@google.com

**Comment by tjaychen** [comment]
@arnonsha @msfschaffner 
I believe the dft side had a similar issue, and a flag was added to generate it appropriately.
so i think the main thing we need to add is some kind of plumbing to inform the dv testbench which vmem to pull (after adding the second vmem generation as a meson target).

@cindychip / @sriyerg what do you guys think? What sort of flag makes sense from dv side to separate foundry and open sims?

**Comment by arnonsha** [comment]
Yes, @msfschaffner added the following option for the closed source OTP:
--data-perm "[15:0],[23:16]"

**Comment by sriyerg** [comment]
My preferred approach is to create a class abstraction for OTP that will use the existing mem bkdr util (already works in both, open and closed source) that will provide a fine-grained control of overwriting specific regions in OTP, rather than pick up a pre-computed image which is slightly restrictive. 

This is about 1-2 days worth of work. I will get back to you on this soon.

**Comment by tjaychen** [comment]
o that's right, you did mention this before.
Thanks for the reminder.

**Comment by sriyerg** [comment]
This is now fixed with #9373. You can add the following in the closed source foundry repo's `chip_sim_cfg.hjson`:
```
    gen_otp_images_cmd_opts: ["--data-perm '[15:0],[23:16]'"]  
```

**Comment by sriyerg** [comment]
> use the existing mem bkdr util (already works in both, open and closed source) that will provide a fine-grained control of overwriting specific regions in OTP, rather than pick up a pre-computed image which is slightly restrictive.
> 
> This is about 1-2 days worth of work. I will get back to you on this soon.

This is a longer term fix, which will be enabled in the coming weeks.
